### ISSUE TITLE
Make subscribing to mailchimp/mailpoet optional by selecting acceptance box

### ISSUE THREAD

**Issue by knireis** [pojome/elementor#4469]
I need to create a form and in this form i would like to ask if this person would like to subscribe to my newsletter.

At the moment i need to make 2 forms to make this possible. An extra form just to subscribe, this is not convenient.

Regards

**Comment by bainternet** [comment]
So basically we are talking about conditional actions?

**Comment by black-eye** [comment]
@bainternet Yes. General conditions would definitely solve this problem and would be more universal (but I suppose more complex to implement). If universal conditions are on the way, excellent. If not, at least these special conditions would be helpful. 

Not sure which solution would be easier and quicker. But you're right - best solution are of course universal contitional actions.

**Comment by DavidCaum** [comment]
+1

**Comment by samuelcerezo** [comment]
+1

**Comment by slewisma** [comment]
This came up from someone else today on Facebook. It seems like the most flexible solution would be a way to make individual actions after submit conditional on specific field values. That would also allow some simple branching. This will quickly lead to more requirements such as setting mailchimp group(s) based on checkboxes checked, etc. so it should be thought through carefully. It is also related to the common request of progressive or multi-page forms where one field's value triggers the visibility of the next set of fields.

**Comment by parfilov** [comment]
+1

**Comment by shindeiru** [comment]
Any chance on this conditional?

**Comment by envyyycreative** [comment]
+1

**Comment by nickelnext** [comment]
+1

**Comment by antoinelubach** [comment]
+1

**Comment by mfauveau** [comment]
+1

**Comment by barrycrous** [comment]
+1

**Comment by GBallzus** [comment]
+1

**Comment by violetbuild1** [comment]
+1 - Is this still not possible yet?

**Comment by ryanviolet** [comment]
+1 - Is this still not possible yet?

**Comment by as1414** [comment]
+1  ... I'm newer to Elementor itself, overall impressive, was surprised that this wasn't thought about already

**Comment by IlPizza79** [comment]
+1, you can't really talk about "mailchimp integration" if you can't simply add a condition in any form :-|

**Comment by imakeinternet** [comment]
Why is this not a thing yet???

**Comment by nicholaszein** [comment]
Hi there,

### 📅  This request is scheduled to be Migrated to the new [Discussions](https://github.com/elementor/elementor/discussions/categories/feature-request) section.

Please do not duplicate this request.

Once we migrate it, we encourage you to cast your **⬆️ Upvote** on the first comment if you feel it should be considered for our roadmap.

**Comment by lpsct** [comment]
+1

**Comment by brajno** [comment]
I tried with something like this, without luck:

```
add_action( 'elementor_pro/forms/process', function( $record, $handler ) {
    // Get the current form ID
    $form_id = $record->get_form_settings( 'form_id' );

    // Replace with your actual form ID
    if ( 'contactform' !== $form_id ) {
        return;
    }

    // Get the form fields
    $raw_fields = $record->get( 'fields' );

    // Get the value of the subscribe checkbox
    $subscribe_checkbox = isset( $raw_fields['subscribe_checkbox'] ) ? $raw_fields['subscribe_checkbox']['value'] : '';

    // Check if the checkbox is checked
    if ( 'newsletter' !== $subscribe_checkbox ) {
        // Remove the MailerLite action if the checkbox is not checked
        $form_settings = $record->get( 'form_settings' );
        if ( isset( $form_settings['submit_actions'] ) ) {
            $actions = $form_settings['submit_actions'];
            if (($key = array_search('mailerlite', $actions)) !== false) {
                unset($actions[$key]);
            }
            // Update the submit_actions setting within the form_settings array
            $form_settings['submit_actions'] = array_values($actions);
            // Set the updated form settings back to the record
            $record->set( 'form_settings', $form_settings );
        }
    }
}, 10, 2 );

**Comment by luukskeur** [comment]
Is this topic (already) migrated to the discussions thread? It doesn't seem so.
For the more technical people among us, mayble this could help: https://github.com/elementor/elementor/issues/7727#issuecomment-479828153

In addition, I have another good example of why this is a good Feature Request: GDPR compliance.
For example, I want so collect my submission OR I want to send the form data to MailChimp/SendinBlue/ActiveCampaign/Brevo/etc. but I ONLY want this form action triggered if a checkbox is checked. If possible even multiple (2) checkboxes.
